### PR TITLE
fix more reverse sigs race conditions

### PR DESCRIPTION
**Pull request by oconnor663**
r? @patrickxb @maxtaco 

Abandoning the "pass from above" approach in favor of "generate JSON once and then edit it afterwards".

### CODE REVIEW

File: go/engine/team_create.go [hunk 1]
```diff
@@ -93,23 +94,24 @@ func (e *TeamCreateEngine) Run(ctx *Context) (err error) {
...
-	sigBodyAfterReverse, err := me.TeamRootSig(deviceSigningKey, teamSection, creationTime)
-	if err != nil {
-		return err
-	}
+	// Update the team section to include the reverse sig, sign it again, and
+	// make a sigchain-v2-style sig out of it. Doing it this way, instead of
+	// generating it twice with different parameters, makes it less likely to
+	// accidentally capture different global state (like ctime and merkle
+	// seqno).
+	sigBodyAfterReverse := sigBodyBeforeReverse
```
Comment (maxtaco) [inline go/engine/team_create.go]: you need to clone, it's too dangerous otherwise.

**Comment by oconnor663** [comment]
cc @mlsteele

File: go/engine/team_create.go (hunk 1 above)
Comment (oconnor663) [inline go/engine/team_create.go]: [we decided this really only applied to signature verification]

File: go/libkb/kbsig.go [hunk 2]
```diff
@@ -657,17 +652,12 @@ func PerUserKeyProof(me *User,
 	pukSigKID keybase1.KID,
```
Comment (mlsteele) [inline go/libkb/kbsig.go]: remove `reverseSig` comment

**Pull request merged**